theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+theta_sketch_dup_alloc<A>::deserialize(const void* bytes, size_t size,
+                                       uint64_t seed) {
+  ensure_minimum_memory(size, static_cast<size_t>(8));
+  const char* ptr = static_cast<const char*>(bytes);
+  uint8_t preamble_longs;
+  ptr += copy_from_mem(ptr, &preamble_longs, sizeof(preamble_longs));
+  uint8_t serial_version;
+  ptr += copy_from_mem(ptr, &serial_version, sizeof(serial_version));
+  uint8_t type;
+  ptr += copy_from_mem(ptr, &type, sizeof(type));
+  uint8_t lg_nom_size;
+  ptr += copy_from_mem(ptr, &lg_nom_size, sizeof(lg_nom_size));
+  uint8_t lg_cur_size;
+  ptr += copy_from_mem(ptr, &lg_cur_size, sizeof(lg_cur_size));
+  uint8_t flags_byte;
+  ptr += copy_from_mem(ptr, &flags_byte, sizeof(flags_byte));
+  uint16_t seed_hash;
+  ptr += copy_from_mem(ptr, &seed_hash, sizeof(seed_hash));
+
+  check_serial_version(serial_version, SERIAL_VERSION);
+  check_seed_hash(seed_hash, get_seed_hash(seed));
+
+  if (type == update_theta_sketch_dup_alloc<A>::SKETCH_TYPE) {
+    typename update_theta_sketch_dup_alloc<A>::resize_factor rf =
+        static_cast<typename update_theta_sketch_dup_alloc<A>::resize_factor>(
+            preamble_longs >> 6);
+    typedef typename std::allocator_traits<A>::template rebind_alloc<
+        update_theta_sketch_dup_alloc<A>>
+        AU;
+    return unique_ptr(
+        static_cast<theta_sketch_dup_alloc<A>*>(
+            new (AU().allocate(1)) update_theta_sketch_dup_alloc<A>(
+                update_theta_sketch_dup_alloc<A>::internal_deserialize(
+                    ptr, size - (ptr - static_cast<const char*>(bytes)), rf,
+                    lg_cur_size, lg_nom_size, flags_byte, seed))),
+        [](theta_sketch_dup_alloc<A>* ptr) {
+          ptr->~theta_sketch_dup_alloc();
+          AU().deallocate(static_cast<update_theta_sketch_dup_alloc<A>*>(ptr),
+                          1);
+        });
+  } else if (type == compact_theta_sketch_dup_alloc<A>::SKETCH_TYPE) {
+    typedef typename std::allocator_traits<A>::template rebind_alloc<
+        compact_theta_sketch_dup_alloc<A>>
+        AC;
+    return unique_ptr(
+        static_cast<theta_sketch_dup_alloc<A>*>(
+            new (AC().allocate(1)) compact_theta_sketch_dup_alloc<A>(
+                compact_theta_sketch_dup_alloc<A>::internal_deserialize(
+                    ptr, size - (ptr - static_cast<const char*>(bytes)),
+                    preamble_longs, flags_byte, seed_hash))),
+        [](theta_sketch_dup_alloc<A>* ptr) {
+          ptr->~theta_sketch_dup_alloc();
+          AC().deallocate(static_cast<compact_theta_sketch_dup_alloc<A>*>(ptr),
+                          1);
+        });
+  }
+  throw std::invalid_argument("unsupported sketch type " +
+                              std::to_string((int)type));
+}
+
+template <typename A>
+bool theta_sketch_dup_alloc<A>::isEqual(
+    const theta_sketch_dup_alloc<A>& r) const {
+  return this->theta_ == r.theta_ && this->is_empty_ == r.is_empty_;
+}
```

Comment:
alloc "is_equal" is defined this way? comments?

Reply:
`isEqual` has been changed to `is_equal`. For the base class, we just need to compare those two variables. More comparisons are in the derived classes.